conans/test/functional/generators/cmake_find_package_multi_test.py

```diff
@@ -526,3 +526,125 @@ def build(self):
         self.assertIn("component libs: $<$<CONFIG:Debug>:;>;$<$<CONFIG:MinSizeRel>:;>;"
                       "$<$<CONFIG:RelWithDebInfo>:;>;$<$<CONFIG:Release>:system_lib_component;",
                       t.out)
+
+
+@pytest.mark.tool_cmake
+class TestNoNamespaceTarget:
+    """ This test case uses build-modules feature to create a target without a namespace. This
+        target uses targets create by Conan (build_modules are included after Conan targets)
+    """
+
+    conanfile = textwrap.dedent("""
+        import os
+        from conans import ConanFile, CMake
+
+        class Recipe(ConanFile):
+            settings = "os", "compiler", "arch", "build_type"
+            exports_sources = ["src/*", "build-module.cmake"]
+            generators = "cmake"
+
+            def build(self):
+                cmake = CMake(self)
+                cmake.configure(source_folder="src")
+                cmake.build()
+
+            def package(self):
+                self.copy("*.h", dst="include", src="src")
+                self.copy("*.lib", dst="lib", keep_path=False)
+                self.copy("*.dll", dst="bin", keep_path=False)
+                self.copy("*.dylib*", dst="lib", keep_path=False)
+                self.copy("*.so", dst="lib", keep_path=False)
+                self.copy("*.a", dst="lib", keep_path=False)
+                self.copy("build-module.cmake", dst="share/cmake")
+
+            def package_info(self):
+                self.cpp_info.libs = ["library"]
+                module = os.path.join("share", "cmake", "build-module.cmake")
+                self.cpp_info.build_modules['cmake_find_package'] = [module, ]
+                self.cpp_info.build_modules['cmake_find_package_multi'] = [module, ]
+    """)
+
+    build_module = textwrap.dedent("""
+        message(">> Build-module is included")
+
+        if(NOT TARGET nonamespace)
+            add_library(nonamespace INTERFACE IMPORTED)
+            target_link_libraries(nonamespace INTERFACE library::library)
```

Comment:
It seems that `set_property(TARGET nonamespace PROPERTY INTERFACE_LINK_LIBRARIES library::library)` has wider support across CMake versions:
https://github.com/conan-io/conan-center-index/issues/4369